WordPress/gutenberg
# Use internal context system to apply toolbar variant to toolbar dropdowns

## What?


Needs #51264 to be merged first.

This PR refactors usages of `DropdownMenu` and `Dropdown` across the repository, removing the need for passing explicitly the `variant: "toolbar"` prop — the components are expected to pick the correct visual variant automatically when rendered inside a `Toolbar`.

## Why?


For a few reasons:

- the components' [internal context system](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#context-system) is designed to tackle this kind of necessity, where a component would react in different ways automatically based on the context it's rendered in
- using `popoverProps`  less explicitly allows for cleaner code. I also think that the requirement of "dropdown menus needing to render differently when in a toolbar" should not be achieved by expecting that the consumers of the component explicitly set a prop.

Most importantly, the new version of `DropdownMenu` that is in the works already relies on the components context system for switching visual style. Updating the "legacy" version of the component should ensure a smoother transition to the new component.

## How?


- The `Toolbar` component now uses the components context system to set a `variant` for `DropdownMenu` and `Dropdown` components.
- The `DropdownMenu` and `Dropdown` components now connect to the components context system, look for the `variant` property passed via context, and set it appropriately on the underlying `Popover` component
- Therefore, we can safely remove all of the instances where `variant: "toolbar"` is explicitly set in the block editor, since all dropdowns should now render with the correct style variation automatically.
- Finally, while in the process of removing `variant` props, I've also replaced instances of the `position` prop with the `placement` prop, since the `position` prop is gradually being phased out

## Testing Instructions





In the block editor and in the site editor select a variety of blocks, and interact with the block toolbar. Make sure that every single dropdown rendered in the toolbar has the same visual styles as on `trunk`

## Screenshots or screencast 

![Screenshot 2023-06-06 at 17 34 37](https://github.com/WordPress/gutenberg/assets/1083581/a97b7388-a8f6-47a6-a16b-9f1554dfb560)
